Ical.Net/DataTypes/CalDateTime.cs

```diff
@@ -201,9 +203,15 @@ public CalDateTime(string value, string? tzId = null)
     {
         var serializer = new DateTimeSerializer();
         CopyFrom(serializer.Deserialize(new StringReader(value)) as CalDateTime
-                 ?? throw new InvalidOperationException($"$Failure for deserializing value '{value}'"));
+                 ?? throw new InvalidOperationException($"$Failure when deserializing value '{value}'"));
+
         // The string may contain a date only, meaning that the tzId should be ignored.
-        _tzId = HasTime ? tzId : null;
+        _tzId ??= HasTime ? tzId : null;
```

Comment:
`CopyFrom` is used only once and it will set `_tzId` regardless of the time.
I suggest to delete `CopyFrom` and use `Initialize`.


```suggestion
        var dt = serializer.Deserialize(new StringReader(value)) as CalDateTime
                 ?? throw new InvalidOperationException($"$Failure when deserializing value '{value}'");

        Initialize(dt._dateOnly, dt._timeOnly, dt._tzId);
```

Reply:
Good catch, `CopyFrom`  is redundant.
Initializing should read
```cs
Initialize(dt._dateOnly, dt._timeOnly, dt.IsUtc ? UtcTzId : tzId);
```
to reflect that `UTC` may be set from the deserialized string (appended 'Z') *_or_* from the `tzId` argument. This correction is actually the aim of the PR.